ddtrace/contrib/pytest/plugin.py

```diff
@@ -77,7 +78,15 @@ def pytest_configure(config):
     config.addinivalue_line("markers", "dd_tags(**kwargs): add tags to current span")
 
     if is_enabled(config):
-        Pin(tags=ci.tags(), _config=ddtrace.config.pytest).onto(config)
+        ci_tags = ci.tags()
+        if ci_tags.get(ci.git.REPOSITORY_URL, None):
+            try:
+                repository_name = re.search(r"(.+?).git", ci_tags[ci.git.REPOSITORY_URL].split("/")[-1]).group(1)
```

Comment:
would url parsing be better here?

parse as a url, then grab the basename of the path, then split extension?

Reply:
Good catch, I forgot we have urlparsing 😆

Reply:
For simple operations on URLs I'd advice against urlparse/urlunparse _on hot paths_. Otherwise, if we don't expect this code to have a big impact at runtime then it's totally fine.

Reply:
@P403n1x87 being in `pytest_configure`, this will just be once on startup